plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -0,0 +1,288 @@
+	err := table.Load()
+	if err != nil {
+		e.logger.Error("failed to get table info", "table", table.Name(), "error", err)
+		return err
+	}
+
+	asset, err := e.buildAsset(table)
+	if err != nil {
+		e.logger.Error("failed to build asset", "table", table.Name(), "error", err)
+		return err
+	}
+
+	emit(models.NewRecord(asset))
+	return nil
+}
+
+func (e *Extractor) buildAsset(tableInfo *odps.Table) (*v1beta2.Asset, error) {
+	defaultSchema := "default"
+	schemaName := tableInfo.SchemaName()
+	if schemaName == "" {
+		schemaName = defaultSchema
+	}
+
+	tableURN := plugins.MaxComputeURN(tableInfo.ProjectName(), schemaName, tableInfo.Name())
+
+	schema := tableInfo.Schema()
+	asset := &v1beta2.Asset{
+		Urn:         tableURN,
+		Name:        schema.TableName,
+		Type:        tableInfo.Type().String(),
+		Description: schema.Comment,
+		CreateTime:  timestamppb.New(tableInfo.CreatedTime()),
+		UpdateTime:  timestamppb.New(tableInfo.LastModifiedTime()),
+		Service:     "maxcompute",
+	}
+
+	tableAttributesData := buildTableAttributesData(tableInfo)
+
+	var columns []*v1beta2.Column
+	for i, col := range schema.Columns {
+		columnData := &v1beta2.Column{
+			Name:        col.Name,
+			DataType:    col.Type.ID().String(),
+			Description: col.Comment,
+			IsNullable:  col.IsNullable,
+			Attributes:  utils.TryParseMapToProto(buildColumnAttributesData(&schema.Columns[i])),
+			Columns:     buildColumns(col.Type),
+		}
+		columns = append(columns, columnData)
+	}
+
+	tableData := &v1beta2.Table{
+		Attributes: utils.TryParseMapToProto(tableAttributesData),
+		Columns:    columns,
+		CreateTime: timestamppb.New(tableInfo.CreatedTime()),
+		UpdateTime: timestamppb.New(tableInfo.LastModifiedTime()),
+	}
+
+	table, err := anypb.New(tableData)
+	if err != nil {
+		e.logger.Warn("error creating Any struct", "error", err)
+	}
+	asset.Data = table
+
+	return asset, nil
+}
+
+func buildColumns(dataType datatype.DataType) []*v1beta2.Column {
+	if dataType.ID() != datatype.STRUCT {
+		return nil
+	}
+
+	structType, ok := dataType.(datatype.StructType)
+	if !ok {
+		return nil
+	}
+
+	var columns []*v1beta2.Column
+	for _, field := range structType.Fields {
+		column := &v1beta2.Column{
+			Name:     field.Name,
+			DataType: field.Type.ID().String(),
+			Columns:  buildColumns(field.Type),
+		}
+		columns = append(columns, column)
+	}
+	return columns
+}
+
+func buildTableAttributesData(tableInfo *odps.Table) map[string]interface{} {
+	attributesData := map[string]interface{}{}
+
+	if tableInfo == nil {
+		return attributesData
+	}
+
+	if tableInfo.ProjectName() != "" {
+		attributesData["project_name"] = tableInfo.ProjectName()
+	}
+
+	if tableInfo.SchemaName() != "" {
+		attributesData["schema"] = tableInfo.SchemaName()
+	}
+
+	if tableInfo.ViewText() != "" {
+		attributesData["table_sql"] = tableInfo.ViewText()
+	}
+
+	if tableInfo.ResourceUrl() != "" {
+		attributesData["resource_url"] = tableInfo.ResourceUrl()
+	}
+
```

Comment:
I believe we still need partition_field information since it is P0. We can get from `table.PartitionColumns()` @Mayurjag